src/sidebar/components/test/menu-item-test.js

```diff
@@ -19,125 +25,225 @@ describe('MenuItem', () => {
-      isSubmenuVisible: true,
-      submenu: <div>Submenu content</div>,
+      // The actual icon for the submenu should be shown on the right.
+      assert.equal(iconSpaces.at(1).children().length, 1);
+    });
+
+    it('does not render submenu content if `isSubmenuVisible` is undefined', () => {
+      const wrapper = createMenuItem({});
+      assert.isFalse(wrapper.exists('Slider'));
+    });
+
+    it('shows submenu content if `isSubmenuVisible` is true', () => {
+      const wrapper = createMenuItem({
+        isSubmenuVisible: true,
+        submenu: <div>Submenu content</div>,
+      });
+      assert.equal(wrapper.find('Slider').prop('visible'), true);
+      assert.equal(
+        wrapper.find('MenuKeyboardNavigation').prop('visible'),
+        true
+      );
+      assert.equal(wrapper.find('Slider').children().text(), 'Submenu content');
+    });
+
+    it('hides submenu content if `isSubmenuVisible` is false', () => {
+      const wrapper = createMenuItem({
+        isSubmenuVisible: false,
+        submenu: <div>Submenu content</div>,
+      });
+      assert.equal(wrapper.find('Slider').prop('visible'), false);
+      assert.equal(
+        wrapper.find('MenuKeyboardNavigation').prop('visible'),
+        false
+      );
+
+      // The submenu content may still be rendered if the submenu is currently
+      // collapsing.
+      assert.equal(wrapper.find('Slider').children().text(), 'Submenu content');
     });
-    assert.equal(wrapper.find('Slider').prop('visible'), true);
-    assert.equal(wrapper.find('Slider').children().text(), 'Submenu content');
-  });
 
-  it('hides submenu content if `isSubmenuVisible` is false', () => {
-    const wrapper = createMenuItem({
-      isSubmenuVisible: false,
-      submenu: <div>Submenu content</div>,
+    it('calls `onToggleSubmenu` when the `closeMenu` callback is fired', () => {
+      const fakeOnToggleSubmenu = sinon.stub();
+      const wrapper = createMenuItem({
+        isSubmenuVisible: true,
+        submenu: <div>Submenu content</div>,
+        onToggleSubmenu: fakeOnToggleSubmenu,
+      });
+      act(() => {
+        wrapper
+          .find('MenuKeyboardNavigation')
+          .props()
+          .closeMenu({ key: 'Enter' });
+      });
+      assert.isTrue(fakeOnToggleSubmenu.called);
     });
-    assert.equal(wrapper.find('Slider').prop('visible'), false);
 
-    // The submenu content may still be rendered if the submenu is currently
-    // collapsing.
-    assert.equal(wrapper.find('Slider').children().text(), 'Submenu content');
+    it('calls `onToggleSubmenu` when the right arrow key is pressed', () => {
+      const fakeOnToggleSubmenu = sinon.stub();
+      const wrapper = createMenuItem({
+        isSubmenuVisible: true,
+        submenu: <div>Submenu content</div>,
+        onToggleSubmenu: fakeOnToggleSubmenu,
+      });
+      wrapper.find('.menu-item').simulate('keydown', { key: 'ArrowRight' });
+      assert.isTrue(fakeOnToggleSubmenu.called);
```

Comment:
It looks to me like this test duplicates the "calls the `onToggleSubmenu` callback when pressing arrow right" test above?

Reply:
yes indeed, good catch